packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -98,6 +103,17 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
     status.stop();
   }
 
+  final Directory oldRunnerDirectory = buildDirectory
+    .parent
+    .childDirectory('runner');
+  if (oldRunnerDirectory.existsSync()) {
+    globals.printStatus(
+      'Delete previous build folder ${oldRunnerDirectory.path}. '
+      'New binaries can be found in ${buildDirectory.childDirectory('runner').path}.'
+    );
+    oldRunnerDirectory.deleteSync(recursive: true);
```

Comment:
Ideally this should have the same logic as the `clean` command's deletion:

https://github.com/flutter/flutter/blob/d9cb50e63de79aa71caa97ecb50f8d06596b3aa0/packages/flutter_tools/lib/src/commands/clean.dart#L116-L143

Namely:

1. Show a progress bar
2. If a file in the build directory is in use, show a helpful error message describing how to stop the program using the file

@christopherfujino How would you feel about moving `CleanCommand.deleteFile` to [`FileSystemUtils`](https://github.com/flutter/flutter/blob/d9cb50e63de79aa71caa97ecb50f8d06596b3aa0/packages/flutter_tools/lib/src/base/file_system.dart#L31C7-L31C23)?

Reply:
Hmm, this seems to do a lot more than I would expect a generic method named `deleteFile`. I'm ok with moving htis to FileSystemUtils if we use a more descriptive method name, like `recursivelyDeleteEntity({bool showProgress = false})`

Reply:
After trying to move this, it makes several changes that are not so welcome on this PR. (tests moving, etc).
Would that be acceptable to first use the initial version, and then refactor this in a subsequent PR?